src/Microsoft.SqlTools.ServiceLayer/DacFx/Contracts/DeploymentOptions.cs

```diff
@@ -296,14 +336,66 @@ public void SetOptions(DacDeployOptions options)
             foreach (var deployOptionsProp in deploymentOptionsProperties)
             {
                 var prop = options.GetType().GetProperty(deployOptionsProp.Name);
-
+                // Note that we set excluded object types here since dacfx has this value as null;
                 if (prop != null)
                 {
-                    deployOptionsProp.SetValue(this, prop.GetValue(options));
+                    SetGenericDeployOptionProps(prop, options, deployOptionsProp);
                 }
             }
         }
 
+        /// <summary>
+        /// Sets the Value and Description to all properties
+        /// </summary>
+        /// <param name="prop"></param>
+        /// <param name="val"></param>
+        /// <param name="deployOptionsProp"></param>
+        public void SetGenericDeployOptionProps(PropertyInfo prop, DacDeployOptions options, PropertyInfo deployOptionsProp)
+        {
+            var val = prop.GetValue(options);
+            var attribute = prop.GetCustomAttributes<DescriptionAttribute>(true).FirstOrDefault();
+
+            // DeploymentOptionProperty is a generic type, which requires value type to intitialize 
+            if (val != null && val.GetType() == typeof(System.Int32))
+            {
+                DeploymentOptionProperty<int> setProp = new DeploymentOptionProperty<int>()
+                {
+                    Value = (int)val,
+                    Description = attribute.Description
+                };
+                deployOptionsProp.SetValue(this, setProp);
+            }
+            else if (val != null && val.GetType() == typeof(System.String))
+            {
+                DeploymentOptionProperty<string> setProp = new DeploymentOptionProperty<string>()
+                {
+                    Value = (string)val,
+                    Description = attribute.Description
+                };
+                deployOptionsProp.SetValue(this, setProp);
+            }
+            else if (val != null && val.GetType() == typeof(System.Boolean))
+            {
+                DeploymentOptionProperty<bool> setProp = new DeploymentOptionProperty<bool>()
+                {
+                    Value = (bool)val,
+                    Description = attribute.Description
+                };
+
+                deployOptionsProp.SetValue(this, setProp);
+            }
+            else if (val != null && val.GetType() == typeof(System.Object))
```

Comment:
If the value is `ObjectType[]` then I don't think this is ever going to be true (and assuming that any object is that specific type array doesn't seem correct.

(but with my comment above I don't think you even need all this logic so this is mostly as an FYI)